pallets/dedir/src/lib.rs

```diff
@@ -0,0 +1,456 @@
+		/// Registry Entry Identifier Already Exists
+		RegistryEntryIdAlreadyExists,
+		/// Registry Entry Identifier Does Not Exists
+		RegistryEntryIdDoesNotExist,
+		/// Registry Attributes Storage Upper Bound Breached
+		MaxRegistryStorageOverflow,
+		/// Regsitry Entry Attributes Storage Upper Bound Breached
+		MaxRegistryEntryStorageOverflow,
+		/// State Addition Failed
+		AddingStateFailed,
+		/// State Not Found In Declared Registry
+		StateNotSupported,
+		/// State Deemed Invalid While Converion From UTF8 To String
+		InvalidState,
+		/// State Deemed Inavlid While Conversion To MaxEncodedInputLength
+		StateConversionError,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A new registry has been created.
+		/// \[who, registry_identifier\]
+		CreatedRegistry { who: T::AccountId, registry_id: RegistryIdOf<T> },
+		/// A new registry entry has been created.
+		/// \[who, registry_identifier, registry_entry_identifier\]
+		CreatedRegistryEntry {
+			who: T::AccountId,
+			registry_id: RegistryIdOf<T>,
+			registry_entry_id: RegistryEntryIdOf<T>,
+		},
+		/// State change has been made for existing registry entry.
+		/// \[who, registry_identifier, registry_entry_identifier, new_state\]
+		RegistryEntryStateChanged {
+			who: T::AccountId,
+			registry_id: RegistryIdOf<T>,
+			registry_entry_id: RegistryEntryIdOf<T>,
+			new_state: RegistryStateOf<T>,
+		},
+	}
+
+	#[pallet::call]
+	/// DeDir pallet declaration.
+	impl<T: Config> Pallet<T> {
+		/// A new Registry has been created.
+		///
+		/// This function allows a user to submit a new create registry request.
+		/// The Registry is created along with various metadata, including the
+		/// attributes, additional-states to be supported.
+		///
+		/// # Arguments
+		/// * `origin` - The origin of the call, which should be a signed user in most cases.
+		/// * `registry_id` - Registry Identifier to be associated with the Registry.
+		/// * `attrs` - (Optional) Bounded Vector of RegistryKeys and associated RegistryTypes which
+		///   will be supported.
+		/// * `additional_states` - (Optional) Bounded Vector of Additional States to be supported
+		///   by the Registry.
+		///
+		/// # Errors
+		/// Returns `Error::<T>::RegistryIdAlreadyExists` if the registry identifier
+		/// exists already.
+		/// Returns `Error::<T>::MaxRegistryStorageOverflow` if the Registry Attributes
+		/// storage upper bound has been breached.
+		/// Returns `Error::<T>::AddingStateFailed` if the state addition fails to be added.
+		///
+		/// # Events
+		/// Emits `CreatedRegistry` when a new Registry is created successfully.
+		///
+		/// # Example
+		/// ```
+		/// create_registry(origin, registry_id, attrs, additional_states)?;
+		/// ```
+		#[pallet::call_index(0)]
+		#[pallet::weight({0})]
+		pub fn create_registry(
+			origin: OriginFor<T>,
+			registry_id: RegistryIdOf<T>,
+			attrs: Vec<(RegistryKeyIdOf<T>, RegistrySupportedTypeOf)>,
```

Comment:
The registry entry attributes should also help to identify the lookup fields that should be used for creating entry identifiers

Reply:
With the discussion with @amarts  previously if the identifier is available then the registry_entry attributes would be fetch able.